extension.json

```diff
@@ -1125,8 +1240,30 @@
             "pattern": "^[A-Za-z0-9!#$%&'*+\\-.^_|~]+$"
           },
           "value": {
-            "type": "string",
-            "minLength": 1
+            "anyOf": [
```

Comment:
Instead of overloading the `value` key with two different data structures, we can use "oneOf" at the top of the schema. For example, we can use a schema like the following for the Cookie condition:

```
{
   "$schema":"http://json-schema.org/draft-04/schema#",
   "oneOf":[
      {
         "type":"object",
         "properties":{
            "name":{
               "pattern":"^[A-Za-z0-9!#$%&'*+\\-.^_|~]+$"
            },
            "value":{
               "type":"string",
               "minLength":1
            },
            "valueIsRegex":{
               "type":"boolean"
            }
         },
         "additionalProperties":false,
         "required":[
            "name",
            "value"
         ]
      },
      {
         "type":"object",
         "properties":{
            "cookieName":{
               "pattern":"^[A-Za-z0-9!#$%&'*+\\-.^_|~]+$"
            },
            "cookieValues":{
               "type":"array",
               "items":{
                  "type":"object",
                  "properties":{
                     "value":{
                        "type":"string",
                        "minLength":1
                     },
                     "valueIsRegex":{
                        "type":"boolean"
                     }
                  },
                  "additionalProperties":false,
                  "required":[
                     "value"
                  ]
               },
               "minItems":1
            }
         },
         "additionalProperties":false,
         "required":[
            "cookieName",
            "cookieValues"
         ]
      }
   ]
}
```

I think we should use this pattern for every delegate changed in this PR.